runtime/dart_isolate.h

```diff
@@ -161,57 +162,34 @@ class DartIsolate : public UIDartState {
-  ///                                         NOT the main entrypoint of Dart
-  ///                                         code in the isolate. That is
-  ///                                         specified when making one of the
-  ///                                         `Run` calls.
+  /// @param[in]  platform_configuration      The platform configuration for
+  ///                                         handling communication with the
+  ///                                         framework.
   /// @param[in]  flags                       The Dart isolate flags for this
   ///                                         isolate instance.
+  /// @param[in]  dart_entrypoint             The name of the dart entrypoint
+  ///                                         function to invoke.
+  /// @param[in]  dart_entrypoint_library     The name of the dart library
+  ///                                         containing the entrypoint.
+  /// @param[in]  isolate_configuration       The isolate configuration used to
+  ///                                         configure the isolate before
+  ///                                         invoking the entrypoint.
   /// @param[in]  isolate_create_callback     The isolate create callback. This
   ///                                         will be called when the before the
   ///                                         main Dart entrypoint is invoked in
   ///                                         the root isolate. The isolate is
   ///                                         already in the running state at
   ///                                         this point and an isolate scope is
-  ///                                         current. This callback is made for
-  ///                                         all isolate launches (including
-  ///                                         the children of the root isolate).
+  ///                                         current.
   /// @param[in]  isolate_shutdown_callback   The isolate shutdown callback.
   ///                                         This will be called before the
   ///                                         isolate is about to transition
   ///                                         into the Shutdown phase. The
   ///                                         isolate is still running at this
   ///                                         point and an isolate scope is
-  ///                                         current.  This callback is made
-  ///                                         for all isolate shutdowns
-  ///                                         (including the children of the
-  ///                                         root isolate).
+  ///                                         current.
+  /// @param[in]  engine_context              Engine-owned state which is
+  ///                                         accessed by the root dart isolate.
```

Comment:
Ah that's true, I'm getting ahead of myself with the terminology here -- true that this stuff is mostly passed into engine from the shell.

UIDartState does seem like a fine place for this since it's the major destination and is at the top of the dependency chain. I'll comment again once I've moved it.